server/app/services/mongodb/migrator.rb

```diff
@@ -1,14 +1,15 @@
 require_relative 'migration'
 require_relative 'migration_proxy'
 require_relative '../logging'
+require_relative '../../helpers/wait_helper'
 
 module Mongodb
   class Migrator
     include Logging
-    include DistributedLocks
+    include WaitHelper
 
     LOCK_NAME = 'mongodb_migrate'.freeze
-    LOCK_TIMEOUT = 60
+    LOCK_TIMEOUT = (60 * 5)
```

Comment:
This could have a comment mentioning something about the relation to the puma worker timeout. The `LOCK_TIMEOUT`has to be longer than the puma `worker_timeout` and `worker_boot_timeout`.